File: leetcode/easy/make-array-zero-by-subtracting-equal-amounts/python/make-array-zero-by-subtracting-equal-amounts.py
# Solution for make-array-zero-by-subtracting-equal-amounts (Python)
# Problem: https://leetcode.com/problems/make-array-zero-by-subtracting-equal-amounts/
def minimumOperations(nums):
    unique_non_zero = set()
    for num in nums:
        if num != 0:
            unique_non_zero.add(num)
    return len(unique_non_zero)

if __name__ == '__main__':
    if (minimumOperations([1,5,0,3,5]) == 3):
        print("Test Case 1 Passed")
    else:
        print("Test Case 1 Failed")
    
    if (minimumOperations([0]) == 0):
        print("Test Case 2 Passed")
    else:
        print("Test Case 2 Failed")

File: leetcode/easy/make-array-zero-by-subtracting-equal-amounts/java/MakeArrayZeroBySubtractingEqualAmounts.java
// Solution for make-array-zero-by-subtracting-equal-amounts (Java)
// Problem: https://leetcode.com/problems/make-array-zero-by-subtracting-equal-amounts/
import java.util.HashSet;
import java.util.Set;

public class MakeArrayZeroBySubtractingEqualAmounts {
    public int minimumOperations(int[] nums) {
        Set<Integer> uniqueNonZero = new HashSet<>();
        for (int num : nums){
            if (num != 0){
                uniqueNonZero.add(num);
            }
        }
        return uniqueNonZero.size();
    }
    public static void main(String[] args) {
        System.out.println("Testing Java Solution for make-array-zero-by-subtracting-equal-amounts");
        MakeArrayZeroBySubtractingEqualAmounts solution = new MakeArrayZeroBySubtractingEqualAmounts();
        int[] nums1 = {1, 5, 0, 3, 5};
        int[] nums2 = {0};
        if(solution.minimumOperations(nums1) == 3) {
            System.out.println("Test 1 Passed");
        } else{
            System.out.println("Test 1 Failed");
        }
        
        if(solution.minimumOperations(nums2) == 0) {
            System.out.println("Test 2 Passed");
        } else{
            System.out.println("Test 2 Failed");
        }
    }

}